core/remote_worker/main.cpp

```diff
@@ -0,0 +1,293 @@
+  outcome::result<Config> readConfig(int argc, char **argv) {
+    namespace po = boost::program_options;
+    Config config;
+    struct {
+      std::string miner_repo;
+      std::string repo;
+      bool can_add_piece = false;
+      bool can_precommit1 = false;
+      bool can_precommit2 = false;
+      bool can_commit = false;
+      bool can_unseal = false;
+    } raw;
+
+    po::options_description desc("Fuhon worker options");
+    auto option{desc.add_options()};
+    option("worker-repo", po::value(&raw.repo)->required());
+    option("miner-repo", po::value(&raw.miner_repo));
+    option("worker-api", po::value(&config.api_port)->default_value(3456));
+    option("addpiece", po::value(&raw.can_add_piece)->default_value(true));
+    option("precommit1", po::value(&raw.can_precommit1)->default_value(true));
+    option("precommit2", po::value(&raw.can_precommit2)->default_value(true));
+    option("commit", po::value(&raw.can_commit)->default_value(true));
+    option("unseal", po::value(&raw.can_unseal)->default_value(true));
+
+    po::variables_map vm;
+    po::store(parse_command_line(argc, argv, desc), vm);
+    po::notify(vm);
+    config.repo_path = raw.repo;
+    if (!boost::filesystem::exists(config.repo_path)) {
+      boost::filesystem::create_directories(config.repo_path);
+    }
+    std::ifstream config_file{config.join("config.cfg")};
+    if (config_file.good()) {
+      po::store(po::parse_config_file(config_file, desc), vm);
+      po::notify(vm);
+    }
+
+    OUTCOME_TRYA(config.miner_api,
+                 api::rpc::loadInfo(raw.miner_repo, "STORAGE_API_INFO"));
+    config.tasks = {
+        primitives::kTTFetch, primitives::kTTCommit1, primitives::kTTFinalize};
+
+    if (raw.can_add_piece) {
+      // TODO(ortyomka): [FIL-344] add kTTAddPiece to tasks
+      spdlog::warn("AddPiece function is not available");
+    }
+    if (raw.can_precommit1) {
+      config.tasks.insert(primitives::kTTPreCommit1);
+    }
+    if (raw.can_precommit2) {
+      config.tasks.insert(primitives::kTTPreCommit2);
+    }
+    if (raw.can_commit) {
+      config.need_download = true;
+      config.tasks.insert(primitives::kTTCommit2);
+    }
+    if (raw.can_unseal) {
+      config.tasks.insert(primitives::kTTUnseal);
+    }
+
+    return config;
+  }
+
+  outcome::result<void> main(Config &config) {
+    auto io{std::make_shared<io_context>()};
+    auto scheduler{std::make_shared<libp2p::protocol::AsioScheduler>(
+        *io, libp2p::protocol::SchedulerConfig())};
+
+    auto mapi{std::make_shared<api::StorageMinerApi>()};
+    api::rpc::Client wsc{*io};
+    wsc.setup(*mapi);
+    OUTCOME_TRY(wsc.connect(config.miner_api.first, config.miner_api.second));
+
+    OUTCOME_TRY(version, mapi->Version());
+    // TODO: make version of miner
+    uint64_t miner_api = 0;
+    if (version.api_version != miner_api) {
+      spdlog::error("lotus-miner API version doesn't match: expected: {}",
+                    miner_api);
+      exit(-1);
```

Comment:
EXIT_FAILURE?

Reply:
Also, we may need such function which logs and exits to reuse in other places

Reply:
Fixed